.github/workflows/ci.yml

```diff
@@ -1,60 +1,82 @@
 name: BuildStockBatch Tests
-on: [pull_request]
+on: 
+  push:
+    branches:
+      - develop
+  pull_request:
+    types:
+      - synchronize
+      - opened
 jobs:
-  build:
+  tests:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
         python-version: ['3.8', '3.9', '3.10']
     name: Tests - Python ${{ matrix.python-version }}
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v3
+        with:
+          path: buildstockbatch
+      - uses: actions/checkout@v3
+        with:
+          repository: NREL/resstock
+          path: resstock
+          ref: develop
+      - uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
+      - name: Download weather
+        run: |
+          mkdir weather
+          cd weather
+          wget --quiet https://data.nrel.gov/system/files/156/BuildStock_TMY3_FIPS.zip
```

Comment:
All the testing files use the same batch of weather files. This keeps it from being downloaded 4 times.